pkg/virtctl/expose/expose.go

```diff
@@ -219,10 +225,55 @@ func (o *Command) RunE(cmd *cobra.Command, args []string) error {
 		service.Spec.ExternalIPs = []string{externalIP}
 	}
 
-	// try to create the service on the cluster
-	_, err = virtClient.CoreV1().Services(namespace).Create(service)
+	serverVersion, err := virtClient.DiscoveryClient().ServerVersion()
+	if err != nil {
+		return err
+	}
+
+	// Make a Regex to say we only want numbers
+	reg, err := regexp.Compile("[^0-9]+")
 	if err != nil {
-		return fmt.Errorf("service creation failed: %v", err)
+		return err
+	}
+
+	major, err := strconv.Atoi(reg.ReplaceAllString(serverVersion.Major, ""))
+	if err != nil {
+		return err
+	}
+	minor, err := strconv.Atoi(reg.ReplaceAllString(serverVersion.Minor, ""))
+	if err != nil {
+		return err
+	}
+
+	if major >= 1 && minor >= 20 {
+		//_, err = virtClient.CoreV1().Services(namespace).Create(service)
+		if err != nil {
+			return fmt.Errorf("service creation failed: %v", err)
+		}
+	} else {
+		// convert the Service to unstructured.Unstructured
+		unstructuredService, err := runtime.DefaultUnstructuredConverter.ToUnstructured(service)
+		if err != nil {
+			return err
+		}
+
+		// Convert 'ipFamily' to 'ipFamilies'
+		err = unstructured.SetNestedStringSlice(unstructuredService, []string{string(ipFamily)}, "spec", "ipFamilies")
+		if err != nil {
+			return err
+		}
```

Comment:
I'd also export this into a function, something like `morphServiceCreationRequest` or similar.